File: src/main/java/murphy/software/games/platform2d/move/model/Actor2D.java
package murphy.software.games.platform2d.move.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.Accessors;

@AllArgsConstructor
@NoArgsConstructor
@Accessors(chain = true, fluent = true)
@Getter
@Setter
public class Actor2D {
    Position position;
    Direction2D direction2D;
    Angle2D angle2D;
    double feetPerSec;
}

File: src/main/java/murphy/software/games/platform2d/move/model/Actor2D.kt
package murphy.software.games.platform2d.move.model

class Actor2D (var position2D: Position2D = Position2D(), var angle2D: Angle2D = Angle2D(), var feetPerSec : Double = 0.0) {

    fun changeX(amount: Double) {
        position2D.x = position2D.x + amount
    }

    fun changeY(amount: Double) {
        position2D.y = position2D.y + amount
    }
}